File: orbit.web/interfaces/api/meal-planner/UpdateRecipeRequest.ts
import type { RecipeIngredient, RecipeStep } from './AddRecipeRequest'

export interface UpdateRecipeRequest {
  id: number
  name: string
  description: string
  ingredients: RecipeIngredient[] | null
  steps: RecipeStep[] | null
  prepTimeMinutes: number | null
  cookTimeMinutes: number | null
  servings: number | null
}

File: Orbit.Domain/DTOs/MealPlanner/UpdateRecipeRequest.cs
namespace Orbit.Domain.DTOs.MealPlanner
{
    public class UpdateRecipeRequest
    {
        public required int Id { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public RecipeIngredientDto[]? Ingredients { get; set; }
        public RecipeStepDto[]? Steps { get; set; }
        public int? PrepTimeMinutes { get; set; }
        public int? CookTimeMinutes { get; set; }
        public int? Servings { get; set; }
    }
}
